ksql-streams/src/main/java/io/confluent/ksql/execution/streams/GroupByParamsFactory.java

```diff
@@ -82,16 +93,36 @@ private static LogicalSchema buildSchema(
   private static Object processColumn(
       final int index,
       final ExpressionMetadata exp,
-      final GenericRow row
+      final GenericRow row,
+      final ProcessingLogger logger
   ) {
     try {
       final Object result = exp.evaluate(row);
       if (result == null) {
+        logger.error(
+            EngineProcessingLogMessageFactory.recordProcessingError(
+               String.format(
+                   "Group-by column with index %d resolved to null."
+                       + " The source row will be excluded from the table.",
+                   index),
+                row
+            )
+        );
         LOG.error("Group-by column with index {} resolved to null. "
             + "The source row will be excluded from the table.", index);
       }
       return result;
     } catch (final Exception e) {
+      logger.error(
+          EngineProcessingLogMessageFactory.recordProcessingError(
+              String.format(
+                  "Error calculating group-by column with index %d. "
+                      + "The source row will be excluded from the table: %s",
+                  index, e.getMessage()),
+              e,
+              row
+          )
+      );
```

Comment:
Same again - remove the `LOG.error` below.  - and you can remove the `LOG` field too.